cataclysmbnteam/Cataclysm-BN
# refactor: Move wetness stats to bodypart

## Checklist



### Required

- [x] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [x] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

### Optional



## Purpose of change

Remove uses of old body part enum from `Character`.



## Describe the solution

Causes one tiny issue: clears wetness from old saves. Decided not to convert wetness because losing wetness stat once on save conversion is insignificant.



## Describe alternatives you've considered



## Testing

* Tests pass
* Start game
* Set weather to rain
* Wait for an hour
* The character should get a morale penalty for wetness
* Set weather to sunny and wait for 6 hours
* The morale penalty for wetness should disappear
* Spawn a deep water tile and get into it for few turns
* Get out of the water
* Note current morale penalty for wetness
* Save+load game
* Wait 5 minutes
* Confirm morale penalty for wetness is close to one from before save+load



## Additional context
